packages/jest-config/src/IstanbulCollector.js

```diff
@@ -4,28 +4,36 @@
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
+*
+* @flow
 */
 'use strict';
 
+import type {Path} from 'types/Config';
+
 const istanbul = require('istanbul');
 
 class IstanbulCollector {
 
-  constructor(sourceText, filename) {
-    const instr = new istanbul.Instrumenter();
-    this._coverageDataStore = {};
-    this._instrumentor = instr;
-    this._origSourceText = sourceText;
-    this._instrumentedSourceText = instr.instrumentSync(sourceText, filename);
+  _coverageDataStore: Object;
+
+  constructor() {
+    this._coverageDataStore = Object.create(null);
   }
 
   getCoverageDataStore() {
     return this._coverageDataStore;
   }
 
-  getInstrumentedSource(storageVarName) {
-    return this._instrumentedSourceText + storageVarName + '.coverState=' +
-           this._instrumentor.currentState.trackerVar + ';';
+  getInstrumentedSource(
+    sourceText: string,
+    filename: Path,
+    storageVarName: string,
+  ) {
+    const instrumentor = new istanbul.Instrumenter();
+    const source = instrumentor.instrumentSync(sourceText, filename);
+    const tracker = instrumentor.currentState.trackerVar;
+    return source + storageVarName + '.coverState=' + tracker + ';';
```

Comment:
Basically right now we create one instrumenter per test file, so this just makes sure that we put the coverage info into the $JEST variable so we can pull it out later.

I'm sure you'll rewrite and fix this soon, however :)